classes/view_import.php

```diff
@@ -773,85 +776,99 @@ public function validate_csvcontent() {
-                }
 
-                if (isset($this->environmentheaders[SURVEYPRO_OWNERIDLABEL])) {
-                    if ($col == $this->environmentheaders[SURVEYPRO_OWNERIDLABEL]) {
-                        // The column for userid.
-                        if ($err = $this->is_valid_userid($value)) {
-                            return $err;
-                        } else {
-                            if ($value != $USER->id) {
-                                if (!isset($submissionsperuser[$value])) {
-                                    $submissionsperuser[$value] = 1;
-                                } else {
-                                    $submissionsperuser[$value]++;
-                                }
+                    if (isset($this->environmentheaders[SURVEYPRO_TIMECREATEDLABEL])) {
+                        if ($col == $this->environmentheaders[SURVEYPRO_TIMECREATEDLABEL]) {
+                            // The column for timecreated.
+                            if ($err = $this->is_valid_creationtime($value)) {
+                                return $err;
                             }
                             continue;
                         }
                     }
-                }
 
-                if (isset($this->environmentheaders[SURVEYPRO_TIMECREATEDLABEL])) {
-                    if ($col == $this->environmentheaders[SURVEYPRO_TIMECREATEDLABEL]) {
-                        // The column for timecreated.
-                        if ($err = $this->is_valid_creationtime($value)) {
-                            return $err;
+                    if (isset($this->environmentheaders[SURVEYPRO_TIMEMODIFIEDLABEL])) {
+                        if ($col == $this->environmentheaders[SURVEYPRO_TIMEMODIFIEDLABEL]) {
+                            // The column for timemodified.
+                            if ($err = $this->is_valid_modificationtime($value)) {
+                                return $err;
+                            }
+                            continue;
                         }
+                    }
+
+                    $itemid = $this->columntoitemid[$col];
+                    if (substr($itemid, -$suffixlength) == SURVEYPRO_IMPFORMATSUFFIX) {
+                        // TODO: format should be verified too.
                         continue;
                     }
+
+                    $message = 'I can not find required itemhelper for item ID = '.$this->columntoitemid[$col].' found in column: '.$col;
+                    debugging('Error at line '.__LINE__.' of file '.__FILE__.'. '.$message , DEBUG_DEVELOPER);
+                } else {
+                    $itemhelper = $this->itemhelperinfo[$col]; // The itemhelperinfo of the item in column = $col.
+                    if ($debug) {
+                        echo 'I am at the line '.__LINE__.' of the file '.__FILE__.'<br />';
+                        echo '$itemhelper:';
+                        var_dump($itemhelper);
```

Comment:
Some day we need to get rid of all these echo & var_dump() uses. Instead, debugging(LEVEL, message) should be used and done. For sure this is not acceptable in production code, IMO.